src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StaticPinnableManagedValueMarshaller.cs

```diff
@@ -107,5 +103,18 @@ private IEnumerable<StatementSyntax> GeneratePinningPath(TypePositionInfo info,
                     EmptyStatement());
             }
         }
+
+        public ByValueMarshalKindSupport SupportsByValueMarshalKind(ByValueContentsMarshalKind marshalKind, TypePositionInfo info, StubCodeContext context, [NotNullWhen(false)] out GeneratorDiagnostic? diagnostic)
+        {
+            if (marshalKind is ByValueContentsMarshalKind.In)
+            {
+                diagnostic = new GeneratorDiagnostic.NotSupported(info, context)
+                {
+                    NotSupportedDetails = SR.InAttributeNotSupportedWithoutOutBlittableArray
+                };
+                return ByValueMarshalKindSupport.NotSupported;
+            }
```

Comment:
We can get here for non-array-based marshallers. Any marshaller that has a static `GetPinnableReference` type will get here. We need to limit this to cases where we are allowing the `[In]` and `[Out]` attributes.

Reply:
There is a different details resource string for pinnable types, I can use that instead, or warn on all if we only want to allow the attributes on arrays.

Reply:
If you look at the logic where we create the `CustomTypeMarshaller` generator, we specify if by-value marshalling should be supported. We can update that code to pass in a `ByValueMarshalKindSupportDescriptor` instead of just a `bool` and have this marshaller forward to the `inner` marshaller.